### ISSUE TITLE
UX: Making node ID and project name (ID) optional

### ISSUE THREAD

**Issue by hansfn** [mglaman/drupalorg-cli#53]
Currently all issue and project commands has mandatory ID argument. I think it would be much more user-friendly if the context was used to try set these IDs if arguments are missing:

- Issue commands can check if current branch name starts with "digits-" and use these digits as NID. If  not found and argument missing, abort with friendly error.
- Project commands can check if current directory is inside modules/ (not core/modules) and use directory name as ID. If  not found and argument missing, abort with friendly error.This depends on a fix for #52.

I can write a patch if this seems like a good idea to you too.

**Comment by mglaman** [comment]
🤔 so I think some code does check the branch name if it matches the branch naming, for detecting the ID. Maybe not 😬 

For the project name, I think we could detect if we're within a git repository and check the remote URL. 

* https://git.drupalcode.org/project/MACHINE_NAME.git
* git@git.drupal.org:project/MACHINE_NAME.git

So We could detect the machine name between `project/` and `.git`

**Comment by mglaman** [comment]
In a follow up, once #56 is merged, let's use `git rev-parse --show-toplevel` instead of `getcwd` for finding the Git repository.

I got it from https://stackoverflow.com/questions/957928/is-there-a-way-to-get-the-git-root-directory-in-one-command and works like a charm.

**Comment by hansfn** [comment]
That was neat. As too often the actual Git command is not very user friendly, but it works :-) It fixes #52 but I guess you had that in mind.

Do you want me to make a patch for optional project name or are you in the flow ;-)

**Comment by mglaman** [comment]
Making a new issue and I'll pop open a PR
